frontend/src/components/component-details.ts

```diff
@@ -79,10 +64,29 @@ export class ComponentDetails extends BaseComponent {
       </div>
     `;
 
+    // Assign safe text content and initialize reports
+    this.setSafeTextContent(component);
     // Initialize reports section with current state
     this.renderReports(latestReports.get());
   }
 
+  // Safely assign text content to avoid XSS
```

Comment:
I don't like this approach. This makes to code less readable. Let's maybe see if we could process the response from the API in the client using some hook to escape all dangerous data and keep the elements declarative